bin/run-bench.js

```diff
@@ -74,67 +82,77 @@ class JSONPrinter {
 
 run()
 
-function run() {
+const errorAndExit = (err, message, code) => {
+  console.log(message)
+  console.error(err)
+  process.exit(code)
+}
+
+async function run() {
   const printer = opts.json ? new JSONPrinter() : new ConsolePrinter()
 
-  a.series(
-    [
-      function resolveGlobs(cb) {
-        if (!globs.length) {
-          cb()
-        }
-
-        a.map(globs, glob, function afterGlobbing(err, resolved) {
-          if (err) {
-            console.error('Failed to glob:', err)
-            process.exitCode = -1
-            cb(err)
+  const resolveGlobs = (cb) => {
+    if (!globs.length) {
+      cb()
+    }
+    const afterGlobbing = (err, resolved) => {
+      if (err) {
+        errorAndExit(err, 'Failed to glob', -1)
+        cb(err)
+      }
+      resolved.forEach(function mergeResolved(files) {
+        files.forEach(function mergeFile(file) {
+          if (tests.indexOf(file) === -1) {
+            tests.push(file)
           }
-          resolved.forEach(function mergeResolved(files) {
-            files.forEach(function mergeFile(file) {
-              if (tests.indexOf(file) === -1) {
-                tests.push(file)
-              }
-            })
-          })
-          cb()
         })
-      },
-      function runBenchmarks(cb) {
-        tests.sort()
-        a.eachSeries(
-          tests,
-          function spawnEachFile(file, spawnCb) {
-            const test = path.relative(benchpath, file)
-
-            const args = [file]
-            if (opts.inspect) {
-              args.unshift('--inspect-brk')
-            }
-            const child = cp.spawn('node', args, { cwd: cwd, stdio: 'pipe' })
-            printer.addTest(test, child)
-
-            child.on('error', spawnCb)
-            child.on('exit', function onChildExit(code) {
-              if (code) {
-                spawnCb(new Error('Benchmark exited with code ' + code))
-              }
-              spawnCb()
-            })
-          },
-          function afterSpawnEachFile(err) {
-            if (err) {
-              console.error('Spawning failed:', err)
-              process.exitCode = -2
-              cb(err)
-            }
-            cb()
-          }
-        )
+      })
+      cb() // ambient scope
+    }
+
+    const globbed = globs.map((item) => glob.sync(item))
+    return afterGlobbing(null, globbed)
+  }
+
+  const spawnEachFile = (file, spawnCb) => {
+    const test = path.relative(benchpath, file)
+
+    const args = [file]
+    if (opts.inspect) {
+      args.unshift('--inspect-brk')
+    }
+
+    // / TODO: remove diagnostic --trace-warnings
+    args.push('--trace-warnings')
```

Comment:
Yes--I wanted to see if I could find the source of the max-listeners-exceeded error, in case this approach is what triggered it. It turns out that the benchmark tests exceed the max listeners without my refactoring.